core/modules/path/path.module

```diff
@@ -453,10 +808,875 @@ function path_entity_update(Entity $entity) {
 }
 
 /**
+ * Implements hook_entity_info().
+ */
+function path_entity_info() {
+  $info['redirect'] = array(
+    'label' => t('Redirect'),
+    'base table' => 'redirect',
+    'entity class' => 'Redirect',
+    'controller class' => 'RedirectController',
+    'entity keys' => array(
+      'id' => 'rid',
+      'bundle' => 'type',
+    ),
+    'fieldable' => FALSE,
+    'uuid' => FALSE,
+    'redirect' => FALSE,
+  );
+
+  return $info;
+}
+
+/**
+ *  Create a redirect hash.
+ *  @param
+ *   Redirect entity.
```

Comment:
Could redirect hashes be elaborated here? What are they used for?

Some minor formatting issues here as well.
```
+ *  Create a redirect hash.
+ *
+ * [Explanation of redirect hashes.]
+ *
+ *  @param Redirect $redirect
+ *   The Redirect entity for which a hash should be generated.
```